Reader: Cleanup `FollowingManage` and related components and state

tyxla opened the pull request:
## Proposed Changes

This PR cleans up a bunch of unused components that we can remove as a follow-up to https://github.com/Automattic/wp-calypso/pull/86410#discussion_r1453411836

## Testing Instructions

Verify everything is green and smoke test the Reader.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

tyxla committed:
ReaderListFollowedSites: Use SearchCard directly

tyxla committed:
Reader: Remove unused FollowingManage components

tyxla committed:
Reader: Remove more unused modules and constants

tyxla committed:
Remove unused ReaderRecommendedSites component

tyxla committed:
Reader: Remove unused ReaderListItem component

tyxla committed:
Remove more unused fns

tyxla committed:
Cleanup some unused styles

tyxla committed:
Remove unused QueryReaderRecommendedSites

tyxla committed:
Reader: Remove unused Reader state

jsnajdr commented on client/reader/stream/reader-list-followed-sites/index.jsx:
```diff
@@ -102,7 +102,19 @@ export class ReaderListFollowedSites extends Component {
+						compact
+						pinned={ false }
+						className="following-manage__search-followed"
+						additionalClasses="following-manage__search-followed-input"
```
Seems you removed the `style.scss` where the `.following-manage__search-followed` were defined.

tyxla replied:
Good find! Brought back the necessary styles in b77f089f65790bcffb6d786e902b1ba3af8651df in the stylesheet that's imported in this particular component.

Note that in the removed styles there were more styles with that classname, but they won't be utilized anymore because they were namespace to a classname that is no longer used.

Can you take another look when you get a chance?

tyxla committed:
Bring back the following-manage__search-followed styles

jsnajdr commented on client/reader/lib/feed-display-helper/index.js:
```diff
@@ -58,4 +43,4 @@ const exported = {
 
 export default exported;
 
-export const { formatUrlForDisplay, getFeedTitle, getFeedStreamUrl, getSiteUrl } = exported;
```
Let's convert this module to a normal ES module, i.e., remove the CJS-compat `exported` object. It's not used anywhere, all imports are normal named ES imports.

tyxla replied:
Done in https://github.com/Automattic/wp-calypso/pull/86996/commits/d50632d25ac4c4d161bf74a8ccca5e385fd981bb

jsnajdr commented on client/reader/lib/feed-display-helper/index.js:
```diff
@@ -14,21 +14,6 @@ const exported = {
-		return feedTitle;
-	},
-
 	getFeedStreamUrl: function ( siteData, feedData ) {
```
`getFeedStreamUrl` and `getSiteUrl` are also unused, the only function ever imported from this module is `formatUrlForDisplay`.

tyxla replied:
Good spot, done as part of https://github.com/Automattic/wp-calypso/pull/86996/commits/d50632d25ac4c4d161bf74a8ccca5e385fd981bb

jsnajdr commented on client/state/reader/follows/reducer.js:
```diff
@@ -317,17 +317,6 @@ export const lastSyncTime = ( state = null, action ) => {
-
-	return state;
-};
-
```
I would consider keeping this function. The API for syncing reader follows is still there and is used. The `isReaderFollowsLoading` selector is a reasonable part of that API, although no components are using it currently. Maybe they should!

tyxla replied:
I'd rather disagree; no need to continue loading an unused reducer, it's very likely we'll not be needing this at all in the future.

jsnajdr commented on client/state/reader/site-dismissals/reducer.js:
```diff
@@ -1,25 +0,0 @@
-	}
-
-	return state;
-};
```
The `dismissSite` and `dismissPost` actions are still used, aren't they? But they are fire-and-forget actions that only need to send the REST requests, and they don't need to store any Redux state anymore, is that right?

I'd consider removing all the data-layer abstractions and converting them to simple thunks:
```js
function dismissSite( siteId ) {
  return ( dispatch ) => wpcom.req.post( '/me/dismiss/...' ).then( () => dispatch( successNotice() ) );
}
```

tyxla replied:
Sounds good, but this feels more like work for another PR. I'd be happy to follow up.

jsnajdr reviewed (commented):
Impressive cleanup 🙂

tyxla committed:
Remove unused helper functions, convert to ES module

tyxla commented:
@jsnajdr I've addressed your feedback, can you take one more look?

Note that I'm planning to follow up on cleaning up the data layer bits from the site and post dismissal functionality in another PR.

jsnajdr reviewed (approved):
Looks good now :shipit:
